Updates to cli ux

wilkermichael opened the pull request:
This PR attempts to improve the UX for the `task create` CLI in a number of ways:

1. Make error "categories" and group like errors closer together in code, subsequent statements provided more details about the error. For example a group of errors related to `Error: unable to create request`
2. Make constants for common CLI error types like `Error: unable to create request` or `Error: unable to create client`
3. Make it clearer that the JSON that is returned during the `inspect` phase of the create is the `Request Payload`
4. Add `no or blank` to the accept/reject instructions to provide the user with confidence when rejecting a plan
5. Disabled logging when running the CLI. We want to control the output completely, so we don't want other parts of the code logging to the output

# Sample Output
- These are more for Jake rather than the assigned code reviewer, but feel free to look if you want.
## Error Scenarios and Output
### Error Creating Request
1. No task file provided
```
consul-terraform-sync task create
==> Error: unable to create request
    no task file provided
    For additional help try 'consul-terraform-sync task create --help'
```
2. Unable to read the task file
```
consul-terraform-sync task create -task-file="invalid.hcl"
==> Error: unable to create request
    unable to read task file
    stat invalid.hcl: no such file or directory
```
3. More than 1 task in task file, only 1 task is allowed (example includes two tasks)
```
==> Error: unable to create request
    task file task_multi.hcl cannot contain more than 1 task, contains 2 tasks
```
4. No tasks are found in task file
```
==> Error: unable to create request
    task file task_empty.hcl does not contain a task, must contain at least one task
```
5. Invalid task file
```
==> Error: unable to create request
    task task_with_invalid_vars.hcl is invalid
    open no_exist.hcl: no such file or directory
```

### Error to Create Client
Generic example
```
==> Error: unable to create client
     some error
```

### Error Unable to Generate Plan
Example: Duplicate tasks
```
==> Inspecting changes to resource if creating task 'api-task'...

    Generating plan that Consul Terraform Sync will use Terraform to execute

==> Error: unable to generate plan for 'api-task'
    400 Bad Request: task with name api-task already exists, see logs for more
details (Request ID: 678b0248-4c60-4140-c10f-7b47e8d1fad5)
```

Example: No server
``` 
==> Inspecting changes to resource if creating task 'api-task'...

    Generating plan that Consul Terraform Sync will use Terraform to execute

==> Error: unable to generate plan for 'api-task'
    Post "http://localhost:8558/v1/tasks?run=inspect": dial tcp [::1]:8558:
connect: connection refused
```

### Error Unable to Create
- Occurs after the `run=now`
- This is just an example error, it's hard to get it to fail here but not during the plan
```
==> Error: unable to create 'api-task'
   400 Bad Request: task could not be rendered
```

## New CLI Output Happy Path
- Make it clearer that the JSON that is returned during the `inspect` phase of the create is the `Request Payload`
- Add `no or blank` to the accept/reject instructions to provide the user with confidence when rejecting a plan
- Add note that task creation may take some time

### Cancel Create Task
```
==> Inspecting changes to resource if creating task 'api-task'...

    Generating plan that Consul Terraform Sync will use Terraform to execute

    Request ID: a8c8ff92-9a8a-10f6-fa2c-f1eaadfe797f
    Request Payload:
    {
      "task": {
        "buffer_period": {
          "enabled": true,
          "max": "20s",
          "min": "5s"
        },
        "description": "Writes the service name, id, and IP address to a file",
        "enabled": true,
        "module": "./example-module",
        "name": "api-task",
        "providers": [
          "local"
        ],
        "services": [
          "api"
        ]
      }
    }

    Plan:
module.api-task.local_file.consul_services: Refreshing state... [id=ebbdd1b5e0d0746766552a34120819aef5470b70]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration
and found no differences, so no changes are needed.

==> Creating the task will perform the actions described above.
    Do you want to perform these actions for 'api-task'?
     - This action cannot be undone.
     - Consul-Terraform-Sync cannot guarantee Terraform will perform
       these exact actions if monitored services have changed.

    Only 'yes' will be accepted to approve, enter 'no' or leave blank to reject

Enter a value: no

    Cancelled creating task 'api-task'
```

### Approve Create Task
```
. . .
==> Creating the task will perform the actions described above.
    Do you want to perform these actions for 'api-task'?
     - This action cannot be undone.
     - Consul-Terraform-Sync cannot guarantee Terraform will perform
       these exact actions if monitored services have changed.

    Only 'yes' will be accepted to approve, enter 'no' or leave blank to reject

Enter a value: yes

==> Creating task api-task...
    Note: this can take some time depending on the module size.
    terminating this process will not stop task creation, see CTS logs for more details

==> Task 'api-task' created
    Request ID: 'ac8aa8fe-ab74-950f-ac8c-b715966db1f3'
```

part of #522

wilkermichael commented on command/task_create.go:
```diff
@@ -120,33 +113,45 @@ func (c *taskCreateCommand) Run(args []string) int {
 		return ExitCodeError
 	}
 
+	client, err := c.meta.taskLifecycleClient()
```
I moved the client creation below the other `errCreatingRequest` scenarios

wilkermichael commented on command/task_create.go:
```diff
@@ -97,19 +98,11 @@ func (c *taskCreateCommand) Run(args []string) int {
 		return ExitCodeRequiredFlagsError
 	}
 
-	client, err := c.meta.taskLifecycleClient()
```
I moved the client creation below the other errCreatingRequest scenarios

wilkermichael commented:
@jherschman can you take a look at the scenarios I posted in the description? Thanks!

mkam reviewed (approved):
Looks good to me!

We had discussed changing the error output to preamble to be just the HTTP status (so like `400 Bad Request` instead of `request returned 400 status code with error`), but let me know if you want that handled in this PR or in a separate one! 

Another CLI UX improvement I've been thinking about is bumping the word wrap count on error messages. I've been finding with the longer errors, the current character count is a bit restrictive and makes it hard to read the whole error. But that might just be me! Or our error messages are too long. 😅  Either way, not something you have to do in this PR, just calling it out as it came to mind.

wilkermichael commented:
> 

@mkam I added your suggestion, and modified the examples to reflect the change :)

wilkermichael commented:
> Looks good to me!
> 
> We had discussed changing the error output to preamble to be just the HTTP status (so like `400 Bad Request` instead of `request returned 400 status code with error`), but let me know if you want that handled in this PR or in a separate one!
> 
> Another CLI UX improvement I've been thinking about is bumping the word wrap count on error messages. I've been finding with the longer errors, the current character count is a bit restrictive and makes it hard to read the whole error. But that might just be me! Or our error messages are too long. 😅 Either way, not something you have to do in this PR, just calling it out as it came to mind.

@mkam I added your suggestion, and modified the examples to reflect the change :)

mkam commented on api/task_lifecycle_client.go:
```diff
@@ -166,10 +166,10 @@ func (d *TaskLifecycleHTTPClient) Do(req *http.Request) (*http.Response, error)
-			return nil, fmt.Errorf("request returned %d status code with error: %s",
-				resp.StatusCode, errMsg)
+			return nil, fmt.Errorf("%d %s: %s",
+				resp.StatusCode, http.StatusText(resp.StatusCode), errMsg)
```
I might be wrong about this, but I think there's a field in the `resp` that has the code and the text combined.

mkam replied:
`resp.Status` is what I was thinking of!
https://pkg.go.dev/net/http#Response

wilkermichael replied:
yep, good call! Changed it, it's `resp.Status`

wilkermichael commented:
@jherschman I'm going to merge this PR in because I think we've accomplished some really good updates, but feel free to review again, and we can open a new PR to address

jherschman reviewed (commented):
These changes look great! They are easy to read and understand, and provide clear context and direction to the user when an error occurs.  Also really enjoy the updates to when a user needs to accept or decline an action (love the added context to the user for 'yes' and 'no' actions. 

Let' ship it!
